infra/main.tf

```diff
@@ -352,7 +352,7 @@ resource "google_artifact_registry_repository_iam_member" "artifact_registry_ci_
   repository = google_artifact_registry_repository.repository.name
   location = var.gcp_location
   role = "roles/artifactregistry.reader"
-  member = "serviceAccount:${google_service_account.ci.email}"
+  member = "serviceAccount:${module.ci_gsa_secret.email}"
```

Comment:
So terraform is able to use a resource that is defined later on in the file...

Reply:
ah yes.